ui/src/warp_runner/manager/commands/constellation_commands.rs

```diff
@@ -64,3 +88,184 @@ fn initialize_items(warp_storage: &mut warp_storage) -> Result<uplink_storage, E
+                    Err(error) => log::error!("Error on update thumbnail: {:?}", error),
+                }
+                log::info!("{:?} file uploaded!", &filename);
+            }
+            Err(error) => log::error!("Error when upload file: {:?}", error),
+        }
+    }
+    get_items_from_current_directory(warp_storage)
+}
+
+fn verify_duplicate_name(
+    current_directory: Directory,
+    filename: String,
+    file_pathbuf: PathBuf,
+) -> String {
+    let mut count_index_for_duplicate_filename = 1;
+    let mut new_file_name = filename.clone();
+    let original = filename;
+    loop {
+        if !current_directory.has_item(&new_file_name) {
+            break;
+        }
+        let file_extension = file_pathbuf
+            .extension()
+            .and_then(OsStr::to_str)
+            .map(str::to_string);
+        let file_stem = file_pathbuf
+            .file_stem()
+            .and_then(OsStr::to_str)
+            .map(str::to_string);
+
+        new_file_name = match (file_stem, file_extension) {
+            (Some(file_stem), Some(file_extension)) => {
+                format!("{file_stem} ({count_index_for_duplicate_filename}).{file_extension}")
+            }
+            _ => format!("{original} ({count_index_for_duplicate_filename})"),
+        };
+
+        log::info!("Duplicate name, changing file name to {}", new_file_name);
+        count_index_for_duplicate_filename += 1;
+    }
+    new_file_name
+}
+
+async fn set_thumbnail_if_file_is_image(
+    warp_storage: warp_storage,
+    filename_to_save: String,
+) -> Result<String, Box<dyn std::error::Error>> {
+    let item = warp_storage
+        .current_directory()?
+        .get_item(&filename_to_save)?;
+    let parts_of_filename: Vec<&str> = filename_to_save.split('.').collect();
+
+    let file = warp_storage.get_buffer(&filename_to_save).await?;
+
+    // Guarantee that is an image that has been uploaded
+    ImageReader::new(Cursor::new(&file))
+        .with_guessed_format()?
+        .decode()?;
+
+    // Since files selected are filtered to be jpg, jpeg, png or svg the last branch is not reachable
+    let mime = match parts_of_filename
+        .iter()
+        .map(|extension| extension.to_lowercase())
+        .last()
+    {
+        Some(m) => match m.as_str() {
+            "png" => IMAGE_PNG.to_string(),
+            "jpg" => IMAGE_JPEG.to_string(),
+            "jpeg" => IMAGE_JPEG.to_string(),
+            "svg" => IMAGE_SVG.to_string(),
+            &_ => "".to_string(),
+        },
+        None => "".to_string(),
+    };
+
+    if !file.is_empty() || !mime.is_empty() {
+        let prefix = format!("data:{};base64,", mime);
+        let base64_image = base64::encode(&file);
+        let img = prefix + base64_image.as_str();
+        item.set_thumbnail(&img);
+        Ok(format_args!("{} thumbnail updated with success!", item.name()).to_string())
+    } else {
+        Err(Box::from(Error::InvalidItem))
```

Comment:
suggest logging the reason for the error - either invalid mime or an empty file.